File: hamersapp/src/main/java/nl/ecci/hamers/beers/NewBeerActivity.java
package nl.ecci.hamers.beers;

import android.app.Activity;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.preference.PreferenceManager;
import android.support.v7.app.ActionBar;
import android.support.v7.widget.Toolbar;
import android.widget.EditText;

import com.android.volley.VolleyError;

import org.json.JSONException;
import org.json.JSONObject;

import nl.ecci.hamers.MainActivity;
import nl.ecci.hamers.R;
import nl.ecci.hamers.helpers.NewItemActivity;
import nl.ecci.hamers.loader.Loader;
import nl.ecci.hamers.loader.PostCallback;

import static nl.ecci.hamers.helpers.Utils.getBeer;

public class NewBeerActivity extends NewItemActivity {

    private int beerID;
    private SharedPreferences prefs;
    private EditText beerName;
    private EditText beerPicture;
    private EditText beerKind;
    private EditText beerPercentage;
    private EditText beerBrewer;
    private EditText beerCountry;


    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.beer_new_activity);

        final Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        final ActionBar actionBar = getSupportActionBar();

        prefs = PreferenceManager.getDefaultSharedPreferences(this);

        if (actionBar != null) {
            actionBar.setDisplayHomeAsUpEnabled(true);
            actionBar.setHomeButtonEnabled(true);
        }

        beerName = (EditText) findViewById(R.id.beer_name);
        beerPicture = (EditText) findViewById(R.id.beer_picture);
        beerKind = (EditText) findViewById(R.id.beer_soort);
        beerPercentage = (EditText) findViewById(R.id.beer_percentage);
        beerBrewer = (EditText) findViewById(R.id.beer_brewer);
        beerCountry = (EditText) findViewById(R.id.beer_country);

        beerID = getIntent().getIntExtra(Beer.BEER, -1);
        if (beerID != -1) {
            Beer beer = getBeer(MainActivity.prefs, beerID);
            beerName.setText(beer.getName());
            beerPicture.setText(beer.getImageURL());
            beerKind.setText(beer.getKind());
            beerPercentage.setText(beer.getPercentage());
            beerBrewer.setText(beer.getBrewer());
            beerCountry.setText(beer.getCountry());

            setTitle("Wijzig " + beer.getName());
        }
    }

    public void postItem() {
        String percentage = beerPercentage.getText().toString();

        if (!percentage.contains("%")) {
            percentage = percentage + "%";
        }

        JSONObject body = new JSONObject();
        try {
            body.put("name", beerName.getText().toString());
            body.put("picture", beerPicture.getText().toString());
            body.put("percentage", percentage);
            body.put("country", beerCountry.getText().toString());
            body.put("brewer", beerBrewer.getText().toString());
            body.put("soort", beerKind.getText().toString());
        } catch (JSONException ignored) {
        }

        Loader.postOrPatchData(this, Loader.BEERURL, body, beerID, new PostCallback() {
            @Override
            public void onSuccess(JSONObject response) {
                Intent returnIntent = new Intent();
                returnIntent.putExtra(SingleBeerActivity.beerName, beerName.getText().toString());
                returnIntent.putExtra(SingleBeerActivity.beerKind, beerKind.getText().toString());
                returnIntent.putExtra(SingleBeerActivity.beerPercentage, beerPercentage.getText().toString());
                returnIntent.putExtra(SingleBeerActivity.beerBrewer, beerBrewer.getText().toString());
                returnIntent.putExtra(SingleBeerActivity.beerCountry, beerCountry.getText().toString());
                setResult(Activity.RESULT_OK, returnIntent);
                finish();
            }

            @Override
            public void onError(VolleyError error) {
                disableLoadingAnimation();
            }
        });
    }
}

File: hamersapp/src/main/java/nl/ecci/hamers/beers/NewBeerActivity.kt
package nl.ecci.hamers.beers

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.widget.EditText

import com.android.volley.VolleyError
import kotlinx.android.synthetic.main.beer_new_activity.*

import org.json.JSONException
import org.json.JSONObject

import nl.ecci.hamers.MainActivity
import nl.ecci.hamers.R
import nl.ecci.hamers.helpers.NewItemActivity
import nl.ecci.hamers.helpers.Utils
import nl.ecci.hamers.loader.Loader
import nl.ecci.hamers.loader.PostCallback

class NewBeerActivity : NewItemActivity() {

    private var beerID: Int = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.beer_new_activity)

        initToolbar()

        beerID = intent.getIntExtra(Beer.BEER, -1)
        if (beerID != -1) {
            val beer = Utils.getBeer(MainActivity.prefs, beerID)
            beer_name.setText(beer.name)
            beer_picture.setText(beer.imageURL)
            beer_soort.setText(beer.kind)
            beer_percentage.setText(beer.percentage)
            beer_brewer.setText(beer.brewer)
            beer_country.setText(beer.country)

            title = "Wijzig " + beer.name
        }
    }

    override fun postItem() {
        var percentage = beer_percentage.text.toString()

        if (!percentage.contains("%")) {
            percentage += "%"
        }

        val body = JSONObject()
        try {
            body.put("name", beer_name!!.text.toString())
            body.put("picture", beer_picture.text.toString())
            body.put("percentage", percentage)
            body.put("country", beer_country.text.toString())
            body.put("brewer", beer_brewer.text.toString())
            body.put("soort", beer_soort.text.toString())
        } catch (ignored: JSONException) {
        }

        Loader.postOrPatchData(this, Loader.BEERURL, body, beerID, object : PostCallback {
            override fun onSuccess(response: JSONObject) {
                val returnIntent = Intent()
                returnIntent.putExtra(SingleBeerActivity.beerName, beer_name.text.toString())
                returnIntent.putExtra(SingleBeerActivity.beerKind, beer_soort.text.toString())
                returnIntent.putExtra(SingleBeerActivity.beerPercentage, beer_percentage.text.toString())
                returnIntent.putExtra(SingleBeerActivity.beerBrewer, beer_brewer.text.toString())
                returnIntent.putExtra(SingleBeerActivity.beerCountry, beer_country.text.toString())
                setResult(Activity.RESULT_OK, returnIntent)
                finish()
            }

            override fun onError(error: VolleyError) {
                disableLoadingAnimation()
            }
        })
    }
}